osu.Game.Rulesets.Taiko/Audio/DrumSampleMapping.cs

```diff
@@ -27,20 +28,34 @@ public DrumSampleMapping(ControlPointInfo controlPoints, AudioManager audio)
 
             foreach (var s in samplePoints)
             {
+                var centre = s.GetSampleInfo();
+                var rim = s.GetSampleInfo(SampleInfo.HIT_CLAP);
+
+                // todo: this is ugly
+                centre.Namespace = "taiko";
+                rim.Namespace = "taiko";
+
                 mappings[s.Time] = new DrumSample
                 {
-                    Centre = s.GetSampleInfo().GetChannel(audio.Sample, "Taiko"),
-                    Rim = s.GetSampleInfo(SampleInfo.HIT_CLAP).GetChannel(audio.Sample, "Taiko")
+                    Centre = addDrawableSound(centre),
+                    Rim = addDrawableSound(rim)
                 };
             }
         }
 
+        private SkinnableSound addDrawableSound(SampleInfo rim)

```

Comment:
along with previous comment, rename to `addSound` + rename the `rim` parameter to `sampleInfo`